apps/quick-learn-backend/src/routes/metadata/metadata.service.ts

```diff
@@ -1,40 +1,53 @@
 import { Injectable } from '@nestjs/common';
-import { RoadmapCategoryService } from '../roadmap-category/roadmap-category.service';
-import { CourseCategoryService } from '../course-category/course-category.service';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CourseCategoryEntity, RoadmapCategoryEntity } from '@src/entities';
+import { Repository } from 'typeorm';
 
 @Injectable()
 export class MetadataService {
   constructor(
-    private readonly roadmapCategoryService: RoadmapCategoryService,
-    private readonly courseCategoryService: CourseCategoryService,
+    @InjectRepository(RoadmapCategoryEntity)
```

Comment:
Please refrain from injecting repositories directly into modules. Instead, move these database operations into dedicated service classes, and then inject those services to handle data access. This helps maintain better separation of concerns.